Admin user create dummy password

Snail51 opened the pull request:
Addresses #17 

Upon accessing the admin user creation page, the password fields are automatically filled with a cryptographically random 16-digit hexadecimal value.

This implementation relies on the user's browser being able to execute the standard JavaScript function `crypto.generateRandomValues()`, which according to [https://caniuse.com/getrandomvalues](https://caniuse.com/getrandomvalues), 96.44% of users' browsers can. In the event that the user's browser cannot execute this function, this case is explicitly handled, and the password fields are left blank (and will need to be manually filled).

Rico (and myself) notes that this solution, while adequate, is inferior to using email to set passwords for new users. Once the associated email server is set up, this code will be refactored.

![image](https://github.com/user-attachments/assets/34a9e96b-059b-4c62-a520-8bba8ade7b78)

Snail51 committed:
fix random missing quotation mark

idk where this went or how it wasn't causing an error. maybe the self closing `/>` was catching it

Snail51 committed:
JS user password generation

perfectly adequate solution to address #17 , populates the password fields on the admin user creation page with cryptographically random passwords.

i will revert this is pushed further to handle this server-side. i *really* dont think we should do that when this is so much easier, introduces no dependencies, and works just as well.

Snail51 committed:
provide link to password.link

recommend admins use https://password.link to securely send passwords to users

Snail51 committed:
explicitly verify user can do crypto.getRandomValues

explicit check that the user's browser knows how to execute `crypto.getRandomValues()`. This will be true for 96.44% of browsers. If the user's browser does not know how to do it, the password field is left blank and will need to be filled manually.

Snail51 committed:
rearrange user create form elements

cosmetic changes to improve clarity

Snail51 committed:
improve password hints

drraccoony commented on resources/views/users/partials/create-user-form.blade.php:
```diff
@@ -67,37 +99,18 @@
-                @endforeach
-            </x-select-input>
-            <x-form-validation for="primary_sector_id" />
+            <p>Tip: Use <a href="https://password.link/" style="color:blue">https://password.link/</a> to securely send passwords to users.</p>
```
We're using tailwind, remove inline Styles.
https://tailwindcss.com/docs/text-color

Snail51 replied:
addressed in 74e62cc125ed9cc905ec3298adb1494f0d33e7f1. I am very aware we are using tailwind, I just forget momentarily and do what is easiest.

drraccoony commented on resources/views/users/partials/create-user-form.blade.php:
```diff
@@ -44,20 +51,45 @@
+
         <!-- Password -->
         <div>
             <div style="display: flex; direction: column">
```
Remove inline styles, Remember we're using tailwind.
https://tailwindcss.com/docs/flex

Snail51 replied:
addressed in 74e62cc125ed9cc905ec3298adb1494f0d33e7f1. I am very aware we are using tailwind, I just forget momentarily and do what is easiest.

drraccoony reviewed (changes requested):
Refactor inline style declarations to use tailwind classes.

Snail51 committed:
remove `style=` attributes

addresses https://github.com/minnesota-furs/mnfursvolunteers/pull/18#pullrequestreview-2397621924

i go on autopilot sometimes and use `style=` as its very easy. we are not using `style=`, however, we are using tailwind. I have refactored all instances of `style=` out of the codebase and used tailwind classes instead.
